solver/src/oneinch_solver/api.rs

```diff
@@ -48,6 +48,36 @@ impl Display for Slippage {
     }
 }
 
+/// Parts to split a swap.
+///
+/// This type is generic on the maximum number of splits allowed.
+#[derive(Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub struct Parts<const N: usize>(usize);
+
+impl<const N: usize> Parts<N> {
```

Comment:
So fancy, monsieur Const Generics 😄

Reply:
They are overkill, but I've been DYING to use them :stuck_out_tongue:.